File: FE-System-checkBV01/src/services/chuyenKhoaService.js
import axiosClient from "./axiosClient";

const API = "/api/v1/chuyen-khoa";

export const getAll = () => axiosClient.get(API);
export const getById = (id) => axiosClient.get(`${API}/${id}`);
export const create = (data) => axiosClient.post(API, data);
export const update = (id, data) => axiosClient.put(`${API}/${id}`, data);
export const remove = (id) => axiosClient.delete(`${API}/${id}`);

File: BE-System-checkBV01/src/main/java/com/checkxmlbv01/websitecheckxmlbv01/service/ChuyenKhoa/ChuyenKhoaService.java
package com.checkxmlbv01.websitecheckxmlbv01.service.ChuyenKhoa;
import java.util.List;


import com.checkxmlbv01.websitecheckxmlbv01.domain.DTO.ChuyenKhoa.ChuyenKhoaRequest;
import com.checkxmlbv01.websitecheckxmlbv01.domain.DTO.ChuyenKhoa.ChuyenKhoaResponse;


public interface ChuyenKhoaService {

    // Tạo mới
    ChuyenKhoaResponse create(ChuyenKhoaRequest request);

    // Cập nhật
    ChuyenKhoaResponse update(Long id, ChuyenKhoaRequest request);

    // Lấy tất cả
    List<ChuyenKhoaResponse> getAll();

    // Lấy theo id
    ChuyenKhoaResponse getById(Long id);

    // Xóa
    void delete(Long id);
}
